src/Controller/InvoiceController.php

```diff
@@ -37,39 +35,24 @@ public function index(InvoiceRepository $invoiceRepository, Request $request): R
         $form = $this->createForm(SearchType::class, $searchData);
         $form->handleRequest($request);
 
+        $typePayment = $request->query->get('typePayment');
+
         if ($form->isSubmitted() && $form->isValid()) {
             $searchData->page = $request->query->getInt('page', 1);
-            $invoices = $invoiceRepository->findByInvoiceName($searchData->q);
-
-            return $this->render('invoice/index.html.twig', [
-                'form' => $form->createView(),
-                'invoices' => $invoices
-            ]);
+            $invoices = $invoiceRepository->findByInvoiceNameAndDescription($searchData->q);
+        } elseif ($typePayment) {
+            $invoices = $invoiceRepository->findByStatus($typePayment);
+        } else {
+            $invoices = $invoiceRepository->findAll();
         }
 
         return $this->render('invoice/index.html.twig', [
             'form' => $form->createView(),
-            'invoices' => $invoiceRepository->findAll(),
+            'invoices' => $invoices,
```

Comment:
We should declare $invoices and set it to empty array to cover all cases

Because if we do not enter in any cases, the render could crash

Reply:
pas de crash c'est déjà gérer sur le twig 

<tbody>
				{% for invoice in invoices %}
					<tr>
						<td>{{ invoice.id }}</td>
						<td>{{ invoice.name }}</td>
						<td>{{ invoice.description }}</td>
						<td>{{ invoice.date ? invoice.date|date('Y-m-d') : '' }}</td>
						<td>{{ invoice.typePayment }}</td>
						<td>{{ invoice.price }}</td>
						<td>
							<a href="{{ path('app_invoice_show', {'id': invoice.id}) }}">show</a>
						</td>
					</tr>
				{% else %}
					<tr>
						<td colspan="7">no records found</td>
					</tr>
				{% endfor %}
			</tbody>